org.eclipse.scout.rt.server/src/main/java/org/eclipse/scout/rt/server/servicetunnel/ServiceTunnelService.java

```diff
@@ -63,38 +65,61 @@ public void init() {
     m_contentHandler = createContentHandler();
   }
 
+  // === SERVICE INVOCATION ===
+
   /**
-   * Check the {@link HttpServletRequest} if signature creation needs to be enabled. Default implementation checks
-   * the {@link IdSignatureClientRequestFilter#ID_SIGNATURE_HTTP_HEADER}.
+   * Handles incoming request by given {@link InputStream}, delegates handling of {@link ServiceTunnelResponse} to {@link #handleResponse(ServiceTunnelResponse)} default implementation.
    */
-  protected boolean enableSignature(HttpServletRequest servletRequest) {
-    return Boolean.TRUE.toString().equalsIgnoreCase(servletRequest.getHeader(IdSignatureClientRequestFilter.ID_SIGNATURE_HTTP_HEADER));
+  public Response incomingRequest(InputStream in) {
+    return handleIncomingRequest(in, this::handleResponse);
+  }
+
+  /**
+   * Default handling, writes {@link ServiceTunnelResponse} directly to given {@link OutputStream}.
```

Comment:
Which given `OutputStream`? No `OutputStream` is given to the method. 

Maybe:

```suggestion
   * Default handling, writes {@link ServiceTunnelResponse} directly to the returned {@link Response}.
   * 
   * @see #serializeServiceResponse(OutputStream, ServiceTunnelResponse) 
```

Reply:
done